pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -711,6 +711,35 @@ self: super: {
     dependencies = with super; [ bclose-vim ];
   });
 
+  nvim-typescript = let
+     # node2nix dynamically resolves nodeDependencies
+     nvim-typescript-deps = (callPackage (
+       runCommand "nvim-typescript-rplugin.nix" {
+         buildInputs = [ nodePackages.node2nix ];
+       } ''
+         mkdir -p $out
+         cd ${super.nvim-typescript.src}/rplugin/node/nvim_typescript
+         node2nix --input package.json \
+                  --lock package-lock.json \
+                  --include-peer-dependencies \
+                  --nodejs-10 \
+                  --development \
+                  --output $out/node-packages.nix \
+                  --node-env $out/node-env.nix \
+                  --composition $out/default.nix
+       ''
+     ) {});
+   in
+     super.nvim-typescript.overrideAttrs(old: {
+       buildInputs = [ nodejs ];
```

Comment:
this probably isn't needed, as it won't be available at runtime
```suggestion
```

Reply:
It needs nodejs to run `npm run build` though

Reply:
then it should in `nativeBuildInputs` :)